services/maintenance/maintenance.service.js

```diff
@@ -1,27 +1,56 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const log = require('../../core/server/log')
-
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class Maintenance extends LegacyService {
-  static get category() {
-    return 'other'
-  }
+const { BaseService } = require('..')
 
+module.exports = class Maintenance extends BaseService {
   static get route() {
     return {
       base: 'maintenance',
-      pattern: ':maintained(yes|no)/:year(\\d{4})',
+      pattern: ':maintained/:year(\\d{4})',
```

Comment:
What's the reason for removing `(yes|no)` from the pattern? Is there a reason to accept other values?

Reply:
Oh, I see, there's a test that hits `/maintenance/until end of 2019/2019.svg`

![](https://img.shields.io/maintenance/until%20end%20of%202019/2019.svg)

Kind of strange that we support that!